src/Sentry/Internal/Hub.cs

```diff
@@ -471,12 +473,9 @@ private SentryId CaptureEvent(SentryEvent evt, SentryHint? hint, Scope scope)
         {
             // We get the span linked to the event or fall back to the current span
             var span = GetLinkedSpan(evt) ?? scope.Span;
-            if (span is not null)
+            if (span is not null && span.IsSampled is not false)
```

Comment:
I'm not actually sure if we want the `&& span.IsSampled is not false` bit in this condition. Potentially we always apply the transaction context to the event, even if the transaction is not sampled? What if people are using complex sampling strategies?

That's probably an issue for another PR though. This PR corrects that problem with no context being applied to the event (trace or propagation)... to guarantee the replay_id will be sent.

Reply:
@AbhiPrasad will know the right way here

My understanding is that we always propagate this, so that downstream they know our decision.

Reply:
> Potentially we always apply the transaction context to the event, even if the transaction is not sampled

I assume you're referring to the `trace` context here, which yes we should always apply regardless if there is a sampled transaction.

Reply:
Cool, I'll update accordingly. Thanks @AbhiPrasad !